src/ui/components/TestSuite/suspense/TestEventDetailsCache.ts

```diff
@@ -275,3 +351,233 @@ async function fetchAndCachePossibleDomNode(
 
   testEventDomNodeCache.cacheValue(domNodeDetails, domNodeDetails.point);
 }
+
+async function fetchPlaywrightStepDetails(
+  replayClient: ReplayClientInterface,
+  filteredEvents: RecordingTestMetadataV3.UserActionEvent[],
+  begin: string,
+  end: string,
+  sources: Map<string, Source>
+): Promise<TestEventDetailsEntry[]> {
+  const testResultPoints = filteredEvents.map(e => e.data.timeStampedPoints.beforeStep!);
+
+  const evalResults: RunEvaluationResult[] = [];
+
+  // This is the injected script that Playwright uses to parse selectors and query the DOM.
+  const playwrightInjectedScriptModule: { source: string } = await import("./injectedScriptSource");
+
+  // These arguments and the preload eval string correspond to the Playwright setup logic
+  // in `playwright-core/src/server/dom.ts::injectedScript()`.
+  // We can hard-code several of these values ourselves.
+
+  // Controls injection into window, and some other external pieces
+  const isUnderTest = false;
+  // Playwright can be used from multiple languages
+  const sdkLanguage = "javascript";
+  // data-testid can be customized by the user
+  const testIdAttributeName = "data-testid";
+  // WebKit on Windows needs 5, everything else is 1
+  const rafCountForStablePosition = 1;
+  const browserName = "chromium";
+  const customSelectorEngines: string[] = [];
```

Comment:
Some of this stuff feels kind of scary in terms of the coupling, like `rafCountForStablePosition`

Reply:
I looked through the Playwright source. The `rafCountForStablePosition` value specifically is something where Safari on Windows does something weird, and everything else is the same.  So, Safari+Win has a value of 5, and everything else is 1.